File: lib/domain/repository/time_capsule_repository.dart
import 'package:time_capsule/domain/model/time_capsule.dart';

abstract class TimeCapsuleRepository {
  Future<List<TimeCapsule>> getAllCapsules();
  Future<TimeCapsule?> getCapsuleById(String id);
  Future<List<TimeCapsule>> getUnopenedCapsules();
  Future<List<TimeCapsule>> getUnlockedButNotNotifiedCapsules();
  Future<String> createCapsule(TimeCapsule capsule);
  Future<void> updateCapsule(TimeCapsule capsule);
  Future<void> deleteCapsule(String id);
  Future<void> markAsOpened(String id);
  Future<void> markNotificationScheduled(String id);
}

File: app/src/main/java/com/timecapsule/app/domain/repository/TimeCapsuleRepository.kt
package com.timecapsule.app.domain.repository

import com.timecapsule.app.domain.model.TimeCapsule
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for Time Capsule operations.
 */
interface TimeCapsuleRepository {

    fun getAllCapsules(): Flow<List<TimeCapsule>>

    suspend fun getCapsuleById(id: Long): TimeCapsule?

    suspend fun getUnopenedCapsules(): List<TimeCapsule>

    suspend fun getUnlockedButNotNotifiedCapsules(): List<TimeCapsule>

    suspend fun createCapsule(timeCapsule: TimeCapsule): Long

    suspend fun updateCapsule(timeCapsule: TimeCapsule)

    suspend fun deleteCapsule(id: Long)

    suspend fun markAsOpened(id: Long)

    suspend fun markNotificationScheduled(id: Long)
}
